default/python/AI/ProductionAI.py

```diff
@@ -1402,10 +1402,37 @@ def generate_production_orders():
                 if (not prioritized) and (this_priority == PriorityType.PRODUCTION_INVASION):
                     fo.issueRequeueProductionOrder(production_queue.size - 1, 0)  # move to front
         print
+    update_stockpile_use()
     fo.updateProductionQueue()
     _print_production_queue(after_turn=True)
 
 
+def update_stockpile_use():
+    """Decide which elements in the production_queue will be enabled for drawing from the imperial stockpile.  This
+    initial version simply ensures that every resource group with at least one item on the queue has its highest
+    priority item be stockpile-enabled.
+
+    :return: None
+    """
+    # TODO: Do a priority and risk evaluation to decide on enabling stockpile draws
+    empire = fo.getEmpire()
+    production_queue = empire.productionQueue
+    resource_groups = set(tuple(el.key()) for el in empire.planetsWithAvailablePP)
+    planets_with_group_stockpile_elem = set()
+    for queue_index in range(len(production_queue)):
+        element = production_queue[queue_index]
+        if element.locationID in planets_with_group_stockpile_elem:
+            continue
```

Comment:
Prefer a comment for clarity here: ```# For now, use stockpile only for the highest priority item in a resource group```.

Reply:
Seems to me that a more accurate (at least for the current code) version of that comment is already only a few (ok, 22) lines higher up, in the routine's docstring, where it seems more important.  A repeat isn't seeming warranted to me.